tools/build.py

```diff
@@ -0,0 +1,681 @@
+                        help="Enable the C++ unit tests. onnxruntime_lib_dir must also be provided.")
+    parser.add_argument("--onnxruntime_lib_dir", type=Path,
+                        help="Path to directory containing the pre-built ONNX Runtime library. "
+                             "Required if enable_unit_tests is True.")
+    # ARM options
+    parser.add_argument("--arm", action="store_true",
+                        help="[cross-compiling] Create ARM makefiles. Requires --update and no existing cache "
+                             "CMake setup. Delete CMakeCache.txt if needed")
+    parser.add_argument("--arm64", action="store_true",
+                        help="[cross-compiling] Create ARM64 makefiles. Requires --update and no existing cache "
+                             "CMake setup. Delete CMakeCache.txt if needed")
+    parser.add_argument("--arm64ec", action="store_true",
+                        help="[cross-compiling] Create ARM64EC makefiles. Requires --update and no existing cache "
+                             "CMake setup. Delete CMakeCache.txt if needed")
+
+    # parser.add_argument("--msvc_toolset", help="MSVC toolset to use. e.g. 14.11")
+
+    # Android options
+    parser.add_argument("--android", action="store_true", help="Build for Android")
+    parser.add_argument("--android_abi", default="arm64-v8a", choices=["armeabi-v7a", "arm64-v8a", "x86", "x86_64"],
+                        help="Specify the target Android Application Binary Interface (ABI)")
+    parser.add_argument("--android_api", type=int, default=27, help="Android API Level, e.g. 21")
+    parser.add_argument("--android_home", type=Path, default=os.environ.get("ANDROID_HOME"),
+                        help="Path to the Android SDK.")
+    parser.add_argument("--android_ndk_path", type=Path, default=os.environ.get("ANDROID_NDK_HOME"),
+                        help="Path to the Android NDK. Typically `<Android SDK>/ndk/<ndk_version>")
+
+    # iOS options
+    parser.add_argument("--build_apple_framework", action="store_true",
+                        help="Build a macOS/iOS framework for the ONNXRuntime.")
+
+    parser.add_argument("--ios", action="store_true", help="build for ios")
+    parser.add_argument("--ios_sysroot", default="",
+                        help="Specify the name of the platform SDK to be used. e.g. iphoneos, iphonesimulator")
+    parser.add_argument("--ios_toolchain_file", default=f"{REPO_DIR}/cmake/ortext_ios.toolchain.cmake", type=Path,
+                        help="Path to ios toolchain file. Default is <repo>/cmake/ortext_ios.toolchain.cmake")
+    parser.add_argument("--xcode_code_signing_team_id", default="",
+                        help="The development team ID used for code signing in Xcode")
+    parser.add_argument("--xcode_code_signing_identity", default="",
+                        help="The development identity used for code signing in Xcode")
+    parser.add_argument("--use_xcode", action="store_true",
```

Comment:
could we just use --cmake_generator instead of having an additional option for xcode?